snappi_ixnetwork/trafficitem.py

```diff
@@ -302,277 +578,133 @@ def _configure_options(self):
-        If the headers list is empty then use the traffic generator default stack.
-        """
-        headers = self.adjust_header(headers)
-        ixn_stack = ixn_stream.Stack.find()
-        for i in range(0, len(headers)):
-            header = headers[i]
-            if len(ixn_stack) <= i:
-                stack = self._add_stack(ixn_stream, stack, header)
-            else:
-                stack_type_id = ixn_stack[i].StackTypeId
-                if stack_type_id in self._STACK_IGNORE:
-                    stack = self._add_stack(ixn_stream, stack, header)
-                elif stack_type_id not in TrafficItem._TYPE_TO_HEADER:
-                    stack = self._add_stack(ixn_stream, ixn_stack[i], header)
-                elif (
-                    TrafficItem._TYPE_TO_HEADER[stack_type_id] != header.choice
-                ):
-                    stack = self._add_stack(ixn_stream, ixn_stack[i], header)
-                else:
-                    stack = ixn_stack[i]
-            self._configure_field(stack.Field, header)
-
-        # scan and compare new stack to overcome IxNetwork stack serialization
-        # then remove additional stack
-        if len(headers) == 0:
+    def _configure_packet(self, tr_type, ixn_stack, snappi_packet):
+        if len(snappi_packet) == 0:
             return
-        stacks_to_remove = []
-        ixn_stack = ixn_stream.Stack.find()
-        header_index = 0
-        for i in range(0, len(ixn_stack)):
-            stack_type_id = ixn_stack[i].StackTypeId
-            if stack_type_id in self._STACK_IGNORE:
+        stacks = [{"xpath": s["xpath"]} for s in ixn_stack]
+        stack_names = [
+            s["xpath"].split(" = ")[-1].strip("']").split("-")[0]
+            for s in ixn_stack
+        ]
+        for i, header in enumerate(snappi_packet):
+            if header._choice not in stack_names:
+                ce_path = stacks[0]["xpath"].split(" = ")[0]
+                index = (
+                    stacks[-1]["xpath"]
+                    .split(" = ")[-1]
+                    .strip("']")
+                    .split("-")[-1]
+                )
+                index = "%s-%s" % (header._choice, index)
+                xpath = "%s = '%s']" % (ce_path, index)
+                self._append_header(header, xpath, stacks)
                 continue
-            if len(headers) <= header_index:
-                stacks_to_remove.append(ixn_stack[i])
+            ind = stack_names.index(header._choice)
+            ixn_fields = ixn_stack[ind]["field"]
+            fields = self._configure_stack_fields(ixn_fields, header)
+            stacks[ind]["field"] = fields
+        return stacks
+
+    def _append_header(self, snappi_header, xpath, stacks, add_trailer=True):
+        field_map = getattr(self, "_%s" % snappi_header._choice.upper())
+        stack_name = self._HEADER_TO_TYPE.get(snappi_header._choice)
+        if stack_name is None:
```

Comment:
Not setting default field (packet.get(packet_field_name, with_default=True)). 

Please check performance after adding this one.

Reply:
self._HEADER_TO_TYPE.get ==> this is different "get" method which gets the _HEADER_TO_TYPE dict.
I assume you are pointing at the default header if user doesn't create any header then we shall default it to ethernet?